storage/innobase/lock/lock0lock.cc

```diff
@@ -7304,6 +7305,26 @@ and less modified rows. Bit 0 is used to prefer orig_trx in case of a tie.
         }
         snprintf(buf, sizeof buf, rollback_msg, victim_pos);
         print(buf);
+
+        if (srv_print_all_deadlocks)
+        {
+          long len= ftell(lock_latest_err_file);
+          if (len <= 0)
+            goto captured;
+          deadlock_info= static_cast<char*>(ut_malloc_nokey(
+            static_cast<size_t>(len) + 1));
+          if (!deadlock_info)
+            goto captured;
+          rewind(lock_latest_err_file);
+          size_t deadlock_info_len= fread(deadlock_info, 1,
+                                          static_cast<size_t>(len),
+                                          lock_latest_err_file);
+          ut_ad(deadlock_info_len <= static_cast<size_t>(len));
+          if (deadlock_info[deadlock_info_len - 1] == '\n')
+            --deadlock_info_len;
+          deadlock_info[deadlock_info_len]= '\0';
+        }
+        captured:;
```

Comment:
It seems your patch is wrongly lying inside the for-loop. Move it out, after lock_cancel_waiting_and_release call, and you won't need an extra label